[BUG] public-xar-repo-plugin unable to retrieve package info from version 4

**Describe the bug**

During a build of exist-db the [public-xar-repo-plugin](https://github.com/eXist-db/public-xar-repo-plugin) is used to resolve bundled XAR packages such as monex, eXide and dashboard.
After updating the central instance at exist-db.org to version 4.1.0 those requests no longer resolved.
The public-repo responded with status code 400 - the reason is swallowed by the plugin.

```
[ERROR] Received HTTP 400 when trying to access: http://localhost:8080/exist/apps/public-repo/find?processor=6.3.0-SNAPSHOT&abbrev=dashboard&info=true
```

All other clients (`curl`, `xh`, `dashboard` and `xst` ) _are_ able to retrieve the package information without issues.

**Expected behavior**

All clients to be able to query package information from public-repo v4. It is especially important that 

**To Reproduce**

1. Start an instance of exist-db 6.0.0 or newer (compatible with public-repo v4)
2. checkout a branch of exist-db 
3. change the property `public.xar.repo.uri` to point to your public-repo instance
4. run `mvn clean package`
7. The build fails in the `exist-distribution` step with a message similar to the one above

**Context (please always complete the following information):**
 - OS: N/A
 - eXist-db Version: 6+
 - Java Version: N/A
-  App Version: 4.0.0, 4.1.0

**Additional context**
- How is eXist-db installed? N/A
- Any custom changes in e.g. `conf.xml`? none

--- a/modules/find.xq
+++ b/modules/find.xq
@@ -47,14 +47,19 @@ declare variable $versions-or-version-range :=
         or exists($semver-max)
 ;
 
-(:
- input: "text/html, application/xhtml+xml, application/xml;q=0.9, image/webp, */*;q=0.8"
- output: ("text/html", "application/json", "application/xml", "image/webp", "*/*")
-:)
-declare function local:parse-accept-header($accept as xs:string) as xs:string* {
-    tokenize($accept, ",")
-        ! normalize-space() (: trim value :)
-        ! tokenize(., ";")[1] (: drop q :)
+(:~
+ : Read and split accept header into a list of mime types, if present
+ : q-values are ignored, order is preserved
+ :
+ : input: "text/html, application/xhtml+xml, application/xml;q=0.9, image/webp, */*;q=0.8"
+ : output: ("text/html", "application/json", "application/xml", "image/webp", "*/*")
+ :)
+declare function local:parse-accept-header() as xs:string* {
+    if (request:get-header-names() = "Accept") then (
+        tokenize(request:get-header("Accept"), ",")
+            ! normalize-space() (: trim value :)
+            ! tokenize(., ";")[1] (: drop q :)
+    ) else ()
 };
 
 declare function local:prefers-json($mime-types as xs:string*) as xs:boolean {
@@ -64,7 +69,7 @@ declare function local:prefers-json($mime-types as xs:string*) as xs:boolean {
         exists($json-index) and (empty($xml-index) or $xml-index > $json-index)
 };
 
-declare variable $json-preferred := local:prefers-json(local:parse-accept-header(request:get-header("Accept")));
+declare variable $json-preferred := local:prefers-json(local:parse-accept-header());
 
 declare function local:render-semver-range($semver as xs:string?, $semver-min as xs:string?, $semver-max as xs:string?) as xs:string {
     if (exists($semver)) then (

